app/javascript/src/utils/compiler.js

```diff
@@ -137,6 +138,93 @@ function extractAndInsertMixins(joinedItems) {
   return joinedItems
 }
 
+/**
+ * @param {string} source
+ * @param {number} fromIndex
+ * @returns
+ */
+function findMatchingClosingBracketIndex(source, fromIndex) {
+  let result = -1
+  let currentDepth = 1
+  for (let i = fromIndex; i < source.length; i++) {
+    const char = source[i]
+    if (char === "{") {
+      currentDepth++
+    } else if (char === "}") {
+      currentDepth--
+      if (currentDepth <= 0) {
+        result = i
+        break
+      }
+    }
+  }
+  return result
+}
+
+/**
+ * @param {string} source
+ * @returns {string}
+ */
+function evaluateConditionals(source) {
+  const conditionalStartRegex = /@if *\( *((?:.|\n)+?) *(==|!=) *((?:.|\n)+?)\) *[ \n]*\{/g
+  const conditionalElseStartRegex = / *@else *\{/
```

Comment:
One thing these 2 regexes miss is different formatting for the brackets. By default the Workshop puts brackets on the next line like
```
if (...)
{
  ...
}
else
{
  ...
}
```

I personally don't prefer that formatting at all, but we should support it. Perhaps something like `@else\s*\{`